### DISCUSSION TITLE
How to handle Passwords in scripts ?

### DISCUSSION THREAD

**Discussion by racklking** [jhuckaby/Cronicle#586] [Q&A]
Hello all, I am writing some scripts to be executed by Cronicle where I need to provide passwords to connect to the systems.
I would not like the password to be visible to anyone, not in UI, not in files.
How can I handle this ? Is there a parameter type password which is masked ?
Sorry I am a newbie.
Thanks for an answer in advance.

**Comment by jhuckaby** [comment]
Hello, Cronicle v1 doesn't really have a secrets manager, but it does have hidden Plugin params.  These are not visible in the UI, but they are sent to your events as JSON and environment variables, so savvy developer can discover them, and they may also be logged, depending on the debug log level.

So this is not a "secrets manager" solution, but it can hide things from the UI.  Create a new Plugin (or copy the Shell Plugin, if you are using that), and add a parameter that has a type "hidden".

https://github.com/jhuckaby/Cronicle/blob/master/docs/WebUI.md#plugin-parameters
